Content.Server/Body/Systems/BodySystem.cs

```diff
@@ -136,22 +130,17 @@ public override HashSet<EntityUid> GibBody(EntityUid bodyId, bool gibOrgans = fa
 
         _audio.Play(body.GibSound, filter, coordinates, true, audio);
 
-        var containers = GetBodyContainers(bodyId, body: body).ToList();
-
-        foreach (var container in containers)
+        foreach (var entity in gibs)
         {
-            foreach (var entity in container.ContainedEntities)
+            if (deleteItems)
+            {
+                QueueDel(entity);
+            }
+            else
             {
-                if (deleteItems)
-                {
-                    QueueDel(entity);
-                }
-                else
-                {
-                    container.Remove(entity, EntityManager, force: true);
-                    SharedTransform.SetCoordinates(entity,coordinates);
-                    entity.RandomOffset(0.25f);
-                }
+                var random = IoCManager.Resolve<IRobustRandom>();
```

Comment:
just add the dependency at the top with the other ones:
i.e.
`[Dependency] private readonly IRobustRandom _random = default!;`

Reply:
Done.